src/ucp/core/ucp_mm.c

```diff
@@ -437,6 +437,29 @@ static void ucp_memh_cleanup(ucp_context_h context, ucp_mem_h memh)
     ucs_free(memh);
 }
 
+ucs_status_t ucp_memh_register_one(ucp_context_h context, ucp_mem_h memh,
+                                   void *address, size_t length,
+                                   ucp_md_index_t md_index,
+                                   uct_md_mem_reg_params_t *reg_params)
+{
+    size_t reg_length, reg_align;
+    void *reg_address, *end;
+
+    if (context->rcache == NULL) {
+        reg_align   = ucs_max(context->tl_mds[md_index].attr.reg_alignment, 1);
+        reg_address = ucs_align_down_pow2_ptr(address, reg_align);
+        end         = UCS_PTR_BYTE_OFFSET(address, length);
+        end         = ucs_align_up_pow2_ptr(end, reg_align);
+        reg_length  = UCS_PTR_BYTE_DIFF(reg_address, end);
+    } else {
+        reg_address = address;
```

Comment:
I would add UCX function like:
ucs_align_ptr_range(void **address_p, size_t *length_p, size_t alignment)

And then do this before calling uct_md_mem_reg_v2 (in ucp_memh_register):
```c
if (rcache == NULL) {
  reg_align = ucs_max(context->tl_mds[md_index].attr.reg_alignment, 1);
  ucs_align_ptr_range(&address, &length, reg_align);
}
```